cosmo_tester/test_suites/cluster/conftest.py

```diff
@@ -154,21 +266,101 @@ def _get_hosts(cfy, ssh_key, module_tmpdir, attributes, logger,
+                },
                 'rabbitmq': {
                     'ca_path': '/tmp/cluster.ca',
                     'cluster_members': rabbit_nodes,
                 },
                 'services_to_install': ['manager_service'],
             }
 
-            if len(dbs) == 0:
+            if high_security:
+                node_cert = cert_base.format(node_type='manager',
+                                             num=node_num,
+                                             extension='crt')
+                node_key = cert_base.format(node_type='manager',
+                                            num=node_num, extension='key')
+
+                util.generate_ssl_certificate(
+                    [node.friendly_name, node.hostname,
+                     node.private_ip_address,
+                     node.ip_address],
+                    node.hostname,
+                    node_cert,
+                    node_key,
+                    ca_cert,
+                    ca_key,
+                )
+
+                node.put_remote_file(
+                    local_path=node_cert,
+                    remote_path='/tmp/node.crt',
+                )
+                node.put_remote_file(
+                    local_path=node_key,
+                    remote_path='/tmp/node.key',
+                )
+                node.additional_install_config['ssl_inputs'] = {
+                    'external_cert_path': '/tmp/node.crt',
+                    'external_key_path': '/tmp/node.key',
+                    'internal_cert_path': '/tmp/node.crt',
+                    'internal_key_path': '/tmp/node.key',
+                    'ca_cert_path': '/tmp/cluster.ca',
+                    'external_ca_cert_path': '/tmp/cluster.ca',
+                }
+                node.additional_install_config['manager']['security'][
+                    'ssl_enabled'] = True
+
+            if len(dbs) > 1:
+                node.additional_install_config['postgresql_server'] = {
+                    'ca_path': '/tmp/cluster.ca',
+                    'cluster': {
+                        'nodes': [str(db.private_ip_address) for db in dbs],
+                    },
+                }
+                node.additional_install_config['postgresql_client'] = {
+                    'server_username': 'postgres',
+                    'server_password': dbs[0].pg_password,
+                }
+                if high_security:
+                    node.additional_install_config['postgresql_client'][
+                        'ssl_client_verification'] = True
+                    node.additional_install_config['postgresql_client'][
+                        'ssl_enabled'] = True
+                    node.additional_install_config['ssl_inputs'][
+                        'postgresql_client_cert_path'] = '/tmp/node.crt'
+                    node.additional_install_config['ssl_inputs'][
+                        'postgresql_client_key_path'] = '/tmp/node.key'
+            elif len(dbs) == 1:
```

Comment:
this check could be more up front, before doing actual things like creating certs etc